### ISSUE TITLE
OSX: linker option --gc-sections not supported

### ISSUE THREAD

**Issue by smlng** [RIOT-OS/RIOT#5787]
The linker on OSX does not support `--gc-sections` as introduced by commit 5fcb09f6. Alternative would be `-dead_strip` on OSX.

**Comment by gebart** [comment]
Could someone else take over this assignment? I don't have an OSX machine to work on. @thomaseichinger, @emmanuelsearch, @kYc0o perhaps?

**Comment by kYc0o** [comment]
I cannot find the commit, could you give the PR number?

**Comment by thomaseichinger** [comment]
@kYc0o I think it should've been #5742.

**Comment by smlng** [comment]
its 5fcb09f

**Comment by kYc0o** [comment]
Confirmed.

```
ld: unknown option: --gc-sections
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [all] Error 1
```

**Comment by kYc0o** [comment]
@smlng can you provide a PR using `-dead_strip`?

**Comment by smlng** [comment]
@kYc0o: well I could add some OS (or linker) specific `#ifdef` magic. However, I've to admit I'm not very familiar with those linker flags and optimizations; I'll check back with a colleague tomorrow who is running OS X, too.

**Comment by smlng** [comment]
Does anyone know (comparing binaries compiled w/ and w/o), how much we actual gain (or reduce size) using this linker option?

**Comment by gebart** [comment]
I don't think code size matters that much on native, but keeping the build
as close to the real targets is quite desirable IMHO.

**Comment by smlng** [comment]
Nah, unsure about that - I mean, compiling for native target will use system/OS specific libs, and those are likely larger (not optimized) compared to the embedded ones, or not? So comparing native target with any real board binary is kind of arbitrary/ambiguous anyway ...

**Comment by smlng** [comment]
Btw. [here](http://blog.senthilthecoder.com/post/lto-linker-commandline/) an interesting post on compiler and linker flags/options when using LTO. Maybe we should apply/integrate this as well?

**Comment by smlng** [comment]
made a PR #5794

**Comment by gebart** [comment]
I believe this was fixed by #5794, reopen if this is not the case.
